lib/ServiceEmitter.js

```diff
@@ -98,6 +107,11 @@ class ServiceEmitter extends EventEmitter {
           this.provider.on('error', err => this.onProviderError(err));
         }
 
```

Comment:
`this.interval();` should be called too, may be making a private interval callback that wraps these two actions.

```javascript
/**
 * Interval callback for {@link this._intervalObject}.
 *
 * @private
 */
_intervalCallback () {

  // If `this._autoRequest` is true and provider is registered then
  // request data to provider.
  if (this._autoRequest && this.isProviderRegistered()) {
    this.provider.requestData();
  }

  // Call `this.interval()` for custom emitters to know.
  this.interval();
}
```

This way we can do:

```javascript
if (this._intervalObject === undefined) {
  this._intervalObject = setInterval(() => this._intervalCallback(), this._delay);
  
  // Call interval callback NOW if `runImmediately` option is true.
  if (this._runImmediately) {
    this._intervalCallback();
  }
}
```